# Lesson 2: Simple Python Data

## Introduction

This chapter introduces several basic building blocks of Python programs:

• literals, like numbers and character strings

• operators, like + and *

• function calls, which take values as inputs and compute new values

• variables, which save values so they can be used later in the program

These are the basic building blocks that are assembled to create programs that you
interact with everyday—from the software running on your smartwatch, to the
infrastructure behind the largest websites, and every app running on your phone.

## Activity

1. An algorithm is:

    A. A step by step sequence of instructions that if followed exactly will solve the problem under consideration.
    

2. Source code is another name for:

    A. the instructions in a program, written in a high-level language.
    

3. What is the difference between a high-level programming language and a lowlevel programming language?

    C. It is high-level if the program must be processed before it can run, and lowlevel if the computer can execute it without
    additional processing.


4. Pick the best replacements for 1 and 2 in the following sentence: When comparing compilers and interpreters, a compiler is like 1 while an interpreter is like 2.

    B. 1 = translating an entire book, 2 = translating a line at a time
    

5. The activecode interpreter allows you to (select all that apply):

    A. save programs and reload saved programs.
    
    B. type in Python source code.
    
    C. execute Python code right in the text itself within the web browser
    
    

6. Codelens allows you to (select all that apply):

    B. control the step by step execution of a program.
    D. execute the Python code that is in codelens.


7. A program is:

    A. a sequence of instructions that specifies how to perform a computation.


8. The differences between natural and formal languages include:

    B. ambiguity, redundancy, and literalness.


9. True or False: Reading a program is like reading other kinds of text.

    B. False


10. The print function:

    B. displays a value on the screen

## Application

### Distance Between Two Points on Earth (27 Lines)
The surface of the Earth is curved, and the distance between degrees of longitude
varies with latitude. As a result, finding the distance between two points on the
surface of the Earth is more complicated than simply using the Pythagorean theorem.
Let (t1, g1) and (t2, g2) be the latitude and longitude of two points on the Earth’s
surface. The distance between these points, following the surface of the Earth, in
kilometers is:

distance = 6371.01
× arccos(sin(𝑡1) × sin(𝑡2) + cos(𝑡1) × cos(𝑡2) × cos(𝑔1 − 𝑔2))

In [10]:
import math
pi = math.pi
latitude1 = float(input('Please enter the latitude of the first point : '))
longitude1 = float(input('Please enter the longitude of the first point: '))
latitude2 = float(input('Please enter the latitude of the second point: '))
longitude2 = float(input('Please enter the longitude of the second point:'))
t1 = latitude1*(pi/180)
g1 = longitude1*(pi/180)
t2 = latitude2*(pi/180)
g2 = longitude2*(pi/180)
r = 6371.01
g = g1-g2
d = r*(math.acos(((math.sin(t1))*(math.sin(t2)))+((math.cos(t1))*(math.cos(t2))*(math.cos(g)))))
print('The distance between the two points is', d, 'km')

Please enter the latitude of the first point : 7.1336
Please enter the longitude of the first point: 124.5307
Please enter the latitude of the second point: 7.0862
Please enter the longitude of the second point:125.6156
The distance between the two points is 119.82389022522179 km


### Based on Google Maps, Midsayap is located at (7.1336° N, 124.5307° E) while the University of Southeastern Philippines is located at (7.0862° N, 125.6156° E) and the distance between those points is approximately 119.828116 km (straight path).